codice/ddf
# DDF-1952: Implemented separate Migratable interfaces for configuration and data

#### What does this PR do?
Provides separate Migratable interfaces for configuration components (mandatory) and data components (optional). 
#### Who is reviewing it?
@lessarderic @figliold @garrettfreibott 
#### How should this be tested?
Aside from a passing Bamboo and PRBS - make sure all bundles start correctly. Ensure `migration:export` still works as expected. 
#### Any background context you want to provide?
This is part of an API that integrators can use to make their modules migrate with the rest of the system. 
#### What are the relevant tickets?
DDF-1952
#### Screenshots (if appropriate)
#### Checklist:

Updated all existing migratables to use new interfaces.
